libs/core/format/include/hpx/modules/format.hpp

```diff
@@ -276,4 +276,36 @@ namespace hpx { namespace util {
         detail::format_to(os, format_str, format_args, sizeof...(Args));
         return os;
     }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail {
+        template <typename Range>
+        struct format_join
+        {
+            Range const& rng;
+            boost::string_ref delim;
+
+            friend void format_value(std::ostream& os, boost::string_ref spec,
+                format_join const& value)
+            {
+                bool first = true;
+                for (auto const& elem : value.rng)
+                {
+                    if (!first)
+                        os << value.delim;
+                    first = false;
+
+                    using value_type = std::decay_t<decltype(elem)>;
+                    detail::formatter<value_type>::call(os, spec, &elem);
+                }
+            }
+        };
+    }    // namespace detail
+
+    template <typename Range>
+    detail::format_join<Range> format_join(
+        Range const& range, boost::string_ref delimiter = "") noexcept
```

Comment:
Absolutely not a blocker, but would `","` be a more appropriate default delimiter? Or not having a default at all?

Reply:
> would `","` be a more appropriate default delimiter?
no

> Or not having a default at all?
sure, go ahead and remove the default